src/providers/ldap/ldap_auth.c

```diff
@@ -234,6 +234,8 @@ static errno_t check_pwexpire_ldap(struct pam_data *pd,
         if (ret != EOK) {
             DEBUG(SSSDBG_CRIT_FAILURE, "pam_add_response failed.\n");
         }
+
+        ret = ERR_PASSWORD_EXPIRED;
```

Comment:
Thank you for the effort.

I have to take a deeper look but as it is now return value from `pam_add_response` is lost (in case of error) and replaced by ERR_PASSWORD_EXPIRED.

Tom

Reply:
Yes, fixed and rebased.